src/ui/states/state-error.svelte

```diff
@@ -1,49 +1,83 @@
 <script lang="ts">
   import { PurchaseFlowError, PurchaseFlowErrorCode } from "../../helpers/purchase-operation-helper";
   import IconError from "../icons/icon-error.svelte";
-  import { onMount } from "svelte";
+  import { getContext, onMount } from "svelte";
   import { type BrandingInfoResponse } from "../../networking/responses/branding-response";
   import { Logger } from "../../helpers/logger.js";
   import MessageLayout from "../layout/message-layout.svelte";
   import { type Product, ProductType } from "../../entities/offerings";
+  import { Translator } from "../localization/translator";
+  import { translatorContextKey } from "../localization/constants";
+  import Localized from "../localization/localized.svelte";
 
   export let brandingInfo: BrandingInfoResponse | null = null;
   export let lastError: PurchaseFlowError;
   export let supportEmail: string | null = null;
   export let productDetails: Product | null = null;
   export let onContinue: () => void;
 
+  const translator: Translator = getContext(translatorContextKey) || Translator.fallback();
+
   onMount(() => {
     Logger.errorLog(
       `Displayed error: ${PurchaseFlowErrorCode[lastError.errorCode]}. Message: ${lastError.message ?? "None"}. Underlying error: ${lastError.underlyingErrorMessage ?? "None"}`,
     );
   });
 
-  function getErrorTitle(): string {
+  function getTranslatedErrorTitle(): string {
     switch (lastError.errorCode) {
       case PurchaseFlowErrorCode.AlreadyPurchasedError:
         if (productDetails?.productType === ProductType.Subscription) {
-          return "Already subscribed";
+          return translator.translate("state_error.error_title_already_subscribed");
```

Comment:
I've been thinking, would it be possible to use some kind of enum for the translation keys instead of plain strings? That way we get compile time safety that there won't be any typos among other benefits. 

If we want to keep the jsons as strings, we can have the enum have an associated string value corresponding to the JSON key and have a test to make sure everything matches in the jsons.

Reply:
We can, I think that now that I figured out all the labels it would be easier to write an enum for this.

The only problem I see is that we need to keep in sync keys in the enum and keys in the files.
I'll add a test for that!